background.ts

```diff
@@ -718,10 +725,10 @@ function newK8sManager() {
     if (state === K8s.State.STOPPING) {
       Steve.getInstance().stop();
     }
-    if (pendingRestart && !backendIsBusy()) {
-      pendingRestart = false;
+    if (pendingRestartContext && !backendIsBusy()) {
```

Comment:
I'd actually prefer `if (pendingRestartContext !== undefined && ...)` because of the boolean-ness of its only property. Making it more explicit makes the code clearer to follow.